sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java

```diff
@@ -201,7 +203,32 @@ public void append(InputStream data, long fileOffset, long length) {
      * @return Information about the created resource.
      */
     public PathInfo flush(long position) {
-        return flushWithResponse(position, false, false, null, null, null, Context.NONE).getValue();
+        return flush(position, false);
+    }
+
+    /**
+     * Flushes (writes) data previously appended to the file through a call to append.
+     * The previously uploaded data must be contiguous.
+     *
+     * <p><strong>Code Samples>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileClient.flush#long-boolean}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure
+     * Docs</a></p>
+     *
+     * @param position The length of the file after all data has been written.
+     * @param overwrite Whether or not to overwrite, should data exist on the file.
+     *
+     * @return Information about the created resource.
+     */
+    public PathInfo flush(long position, boolean overwrite) {
+        DataLakeRequestConditions requestConditions = null;
```

Comment:
Should this instantiate a new `DataLakeRequestConditions`? I'm seeing other places which instantiate instead of leaving it as `null`.

Reply:
Oh whoops thanks for catching that

Reply:
Done